ci/jenkinsfile.groovy

```diff
@@ -98,13 +111,38 @@ stage('Build') {
       node('CPU-SMALL') {
         ws(per_exec_ws('mlc-llm-build')) {
           init_git(true)
-          sh(script: "ls", label: 'debug')
           sh(script: 'git clone --recursive https://github.com/mlc-ai/package', label: 'Clone mlc-ai/package')
           sh(script: "${pkg_cuda} ./ci/task/build_lib.sh", label: 'Build MLC LLM runtime')
           sh(script: "${pkg_cuda} ./ci/task/build_wheel.sh", label: 'Build MLC LLM wheel')
+          pack_lib('mlc_llm_runtime_cuda', 'wheels/*.whl')
           sh(script: "${pkg_cuda} ./ci/task/build_clean.sh", label: 'Cleanups after build')
         }
       }
     }
   )
 }
+
+def compile(target) {
+  node('CPU-SMALL') {
+    ws(per_exec_ws('mlc-llm-compile-models')) {
+      init_git(false)
+      unpack_lib('mlc_llm_runtime_cuda', 'wheels/*.whl')
+      withCredentials([string(
+        credentialsId: 'hf_download_token',
+        variable: 'HF_TOKEN',
+      )]) {
+        sh(
+          // Avoid using "${pkg_cuda} ./ci/task/compile_models.sh cuda ${HF_TOKEN}" for security
+          script: "${pkg_cuda} ./ci/task/compile_models.sh \"${target}\" " + '$HF_TOKEN',
```

Comment:
This is not a huge burden to repetitive download from HuggingFace because we don't download concrete weights, only metadata. And indeed, testing environment has to depend on secret tokens and it is common practice.

However, I would argue that in this particular case, as it's definitely avoidable to repetitively download from HF, and more stable to avoid repetitive download from HF, if we adopt the succinct approach as I mentioned in [my previous comment](https://github.com/mlc-ai/mlc-llm/pull/1615/files#r1456938603).